templates/aws-stack.yml

```diff
@@ -1190,6 +1190,7 @@ Resources:
       Parameters:
         BuildkiteAgentTokenParameter: !If [ UseCustomerManagedParameterPath, !Ref BuildkiteAgentTokenParameterStorePath, !Ref BuildkiteAgentTokenParameter ]
         BuildkiteAgentTokenParameterStoreKMSKey: !If [ UseCustomerManagedKeyForParameterStore, !Ref BuildkiteAgentTokenParameterStoreKMSKey, "" ]
+        RolePermissionsBoundaryARN: !If [ SetInstanceRolePermissionsBoundaryARN, !Ref InstanceRolePermissionsBoundaryARN, "" ]
```

Comment:
would it be worth having a separate permissions boundary for the scaling lambda and and the EC2 instance? it seems like they have separate concerns in terms of what kind of other AWS services they interact with

Reply:
I was wondering that too.
But, I suspect in most/all organizations, a single Permission Boundary will be assigned to a CI stack (or even to the team operating that stack), so maybe nobody would make use of a separate one.

I think we could ship it like this now, and potentially add separate parameters/boundaries for different components if/when asked for. For backwards-compatibility, we could use conditionals to fall back to the existing shared `InstanceRolePermissionsBoundaryARN` if it's present and a specific one isn't. So — we're not painting ourselves into a corner by merging this as-is, I think.

Reply:
☝️this isn't a "please do this" review comment btw, i'm genuinely interested as we don't tend use permissions boundaries much internally.

Reply:
We would only have one boundary condition per deployment.  Having a second condition as an option should be fine - just need a default and conditions for those who opt to only use one boundary condition.